src/hotspot/share/cds/cdsConfig.cpp

```diff
@@ -847,3 +852,22 @@ bool CDSConfig::is_dumping_method_handles() {
 }
 
 #endif // INCLUDE_CDS_JAVA_HEAP
+
+// AOT code generation and its archiving is disabled by default.
+// We enable it only in the final image dump after the metadata and heap are dumped.
+// This affects only JITed code because it may have embedded oops and metadata pointers
+// which AOT code encodes as offsets in final CDS archive regions.
+
+static bool _is_dumping_aot_code_enabled = false;
```

Comment:
Can we rename it to `_is_dumping_aot_code`? IMO the `_enabled` suffix looks redundant here.

Reply:
`_is_dumping_aot_code` means "we are dumping code now" for me.
`_is_dumping_aot_code_enabled` means we are allowed to dump code now. Which is correct because it will be used during attempt to dump nmethods:
```
AOTCodeEntry* AOTCodeCache::store_nmethod(nmethod* nm, AbstractCompiler* compiler, bool for_preload) {
  if (!CDSConfig::is_dumping_aot_code_enabled()) {
    return nullptr; // The metadata and heap in the CDS image haven't been finalized yet.
  }
```